CMakeLists.txt

```diff
@@ -14,21 +14,36 @@ if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(GSL_STANDALONE_PROJECT ON)
 endif ()
 
+
+include(CheckCXXCompilerFlag)
+if (NOT MSVC)
+    CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
+else()
+    CHECK_CXX_COMPILER_FLAG("-std:c++14" COMPILER_SUPPORTS_CXX14)
+endif()
+
 # when minimum version required is 3.8.0 remove if below
 # both branches do exactly the same thing
 if (CMAKE_VERSION VERSION_LESS 3.7.9)
     if (NOT MSVC)
-        include(CheckCXXCompilerFlag)
-        CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
         if(COMPILER_SUPPORTS_CXX14)
             target_compile_options(GSL INTERFACE "-std=c++14")
```

Comment:
I think these branches, that only differ in `=`/`:`, can be merged by changing the string to the following generator expression: `$<$<NOT:$<BOOL:${MSVC}>>:"-std=c++14"> $<$<BOOL:${MSVC}>:"-std:c++14">`.

Reply:
@johelegp yes you are right... will keep as is for now just because I want to separate adding c++17 tests in the next PR and will need to change this code anyway. Will address your comment there.

Reply:
@johelegp: I'd argue that the branches are actually more readable than the equivalent generator expression